core/src/main/java/com/graphhopper/routing/util/parsers/RacingBikePriorityParser.java

```diff
@@ -27,31 +25,33 @@ protected RacingBikePriorityParser(DecimalEncodedValue priorityEnc, DecimalEncod
         preferHighwayTags.add("tertiary_link");
         preferHighwayTags.add("residential");
 
-        avoidHighwayTags.put("motorway", BAD);
-        avoidHighwayTags.put("motorway_link", BAD);
-        avoidHighwayTags.put("trunk", BAD);
-        avoidHighwayTags.put("trunk_link", BAD);
-        avoidHighwayTags.put("primary", AVOID_MORE);
-        avoidHighwayTags.put("primary_link", AVOID_MORE);
+        avoidHighwayTags.put("motorway", 0.5);
+        avoidHighwayTags.put("motorway_link", 0.5);
+        avoidHighwayTags.put("trunk", 0.5);
+        avoidHighwayTags.put("trunk_link", 0.5);
+        avoidHighwayTags.put("primary", 0.6);
+        avoidHighwayTags.put("primary_link", 0.6);
 
         setSpecificClassBicycle("roadcycling");
 
         avoidSpeedLimit = 81;
     }
 
     @Override
-    void collect(ReaderWay way, double wayTypeSpeed, boolean bikeDesignated, TreeMap<Double, PriorityCode> weightToPrioMap) {
-        super.collect(way, wayTypeSpeed, bikeDesignated, weightToPrioMap);
+    double collect(ReaderWay way, double wayTypeSpeed, boolean bikeDesignated) {
+        double prio = super.collect(way, wayTypeSpeed, bikeDesignated);
 
         String highway = way.getTag("highway");
         if ("service".equals(highway) || "residential".equals(highway)) {
-            weightToPrioMap.put(40d, SLIGHT_AVOID);
+            prio *= 0.7;
```

Comment:
I think the factor 0.7 is too small, I would set it to 0.9. In the old code we fixed set it to SLIGHT_AVOID (0.9). Can you please explain your reasoning behind?